### DISCUSSION TITLE
Difference between DDS and RTPS Protocol

### DISCUSSION THREAD

**Discussion by Nitind7** [eProsima/Fast-RTPS-docs#918] [Q&A]
What's difference between DDS and RTPS protocols? Which one is good option when need to implement inter process communication?

**Comment by JesusPoderoso** [comment]
Hi @Nitind7, you may be interested in _[what is DDS?](https://fast-dds.docs.eprosima.com/en/latest/fastdds/getting_started/definitions.html)_ section of the Fast DDS documentation. In summary, the RTPS protocol supports DDS applications. Both DDS and RTPS share the same publication-subscription-topic structure, and allow configuration of their endpoints (QoS and attributes, respectively). 
If the communication from the user application to the final transport is represented in layers, the [Fast DDS architecture](https://fast-dds.docs.eprosima.com/en/latest/fastdds/library_overview/library_overview.html) is structured with the RTPS domain as the inner layer and the DDS domain as the outer layer.

> Which one is good option when need to implement inter process communication?

For inter-process communication, the best delivery mechanism is [data-sharing](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/datasharing.html), but there are [several constraints](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/datasharing.html#constraints) that need to be met to use it. If not possible, then the [shared-memory transport](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html) is the following option.
In both cases, mind that serialization can be avoided (reducing the number of copies) using [loans](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/zero_copy/zero_copy.html). [This thread](https://github.com/eProsima/Fast-DDS/discussions/3654#discussioncomment-6372721) may be of interest to you.
